src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -677,6 +677,11 @@ internal sealed class CacheWriter : IDisposable
 
             private const char RelatedPropertySeparator = ';';
 
+            public CacheWriter(ResolvePackageAssets task, IPackageResolver resolver) : this(task)
```

Comment:
Suggestion: add a comment here that this constructor is for tests.

Reply:
good call - pushed just now.